samples/tooling/profiles/profiles.cpp

```diff
@@ -0,0 +1,711 @@
+	vertex_input_state.pVertexBindingDescriptions           = vertex_input_bindings.data();
+	vertex_input_state.vertexAttributeDescriptionCount      = static_cast<uint32_t>(vertex_input_attributes.size());
+	vertex_input_state.pVertexAttributeDescriptions         = vertex_input_attributes.data();
+
+	VkGraphicsPipelineCreateInfo pipeline_create_info = vkb::initializers::pipeline_create_info(pipeline_layout, render_pass, 0);
+
+	pipeline_create_info.pVertexInputState   = &vertex_input_state;
+	pipeline_create_info.pInputAssemblyState = &input_assembly_state;
+	pipeline_create_info.pRasterizationState = &rasterization_state;
+	pipeline_create_info.pColorBlendState    = &color_blend_state;
+	pipeline_create_info.pMultisampleState   = &multisample_state;
+	pipeline_create_info.pViewportState      = &viewport_state;
+	pipeline_create_info.pDepthStencilState  = &depth_stencil_state;
+	pipeline_create_info.pDynamicState       = &dynamic_state;
+	pipeline_create_info.stageCount          = static_cast<uint32_t>(shader_stages.size());
+	pipeline_create_info.pStages             = shader_stages.data();
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipeline));
+}
+
+// Prepare and initialize uniform buffer containing shader uniforms
+void Profiles::prepare_uniform_buffers()
+{
+	// Vertex shader uniform buffer block
+	uniform_buffer_vs = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                        sizeof(ubo_vs),
+	                                                        VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+	                                                        VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	update_uniform_buffers();
+}
+
+void Profiles::update_uniform_buffers()
+{
+	// Fixed ubo with projection and view matrices
+	ubo_vs.projection = camera.matrices.perspective;
+	ubo_vs.view       = camera.matrices.view;
+
+	uniform_buffer_vs->convert_and_update(ubo_vs);
+}
+
+bool Profiles::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	camera.type = vkb::CameraType::LookAt;
+	camera.set_position(glm::vec3(0.0f, 0.0f, -10.0f));
+	camera.set_rotation(glm::vec3(0.0f));
+
+	// Note: Using reversed depth-buffer for increased precision, so Znear and Zfar are flipped
+	camera.set_perspective(60.0f, (float) width / (float) height, 256.0f, 0.1f);
+
+	generate_textures();
+	generate_cubes();
+	prepare_uniform_buffers();
+	setup_descriptor_set_layout();
+	prepare_pipelines();
+	setup_descriptor_pool();
+	setup_descriptor_set();
+	build_command_buffers();
+	prepared = true;
+	return true;
+}
+
+void Profiles::render(float delta_time)
+{
+	if (!prepared)
+		return;
+	draw();
+}
+
+void Profiles::view_changed()
+{
+	update_uniform_buffers();
+}
+
+// This sample overrides the device creation part of the framework
```

Comment:
wondering if we should move these to the top of the file, as they are the main focus of the sample?

Reply:
That's a good point. I'll move both to the top of the sample, right below the constructor/destructor.